pywinauto/unittests/test_application_macos.py

```diff
@@ -0,0 +1,185 @@
+            self.assertFalse(app.kill(soft=True))
+
+        def test_is_process_running(self):
+            """test is_process_running of the application"""
+            app = Application()
+            app.start(bundle_id='com.yourcompany.send-keys-test-app')
+            self.assertTrue(app.is_process_running())
+            app.kill()
+            app.wait_for_process_exit()
+            self.assertFalse(app.is_process_running())
+
+        def test_start_by_bundle(self):
+            """test start() works correctly after being called by bundle"""
+            app = Application()
+            self.assertEqual(app.ns_app, None)
+            app.start(bundle_id='com.yourcompany.send-keys-test-app')
+            self.assertNotEqual(app.ns_app, None)
+            app.kill()
+
+        def test_process_id(self):
+            """Test that connect_() works with a path"""
+            app = Application()
+            self.assertEqual(app.process_id, None)
+            app.start(bundle_id='com.yourcompany.send-keys-test-app')
+            self.assertNotEqual(app.process_id, None)
+            app.kill()
+
+        def test_cpu_usage(self):
+            """Verify that cpu_usage() works correctly"""
+            app = Application()
+            app.start(bundle_id='com.yourcompany.send-keys-test-app')
+            self.assertEqual(0.0 <= app.cpu_usage() <= 100.0, True)
+            app.kill()
+
+        def test_wait_for_process_running(self):
+            """test is_process_running of the application"""
+            app = Application()
+            self.assertFalse(app.is_process_running())
+            app.start(bundle_id='com.yourcompany.send-keys-test-app')
+            app.wait_for_process_running()
+            self.assertTrue(app.is_process_running())
+            app.kill()
+
+
+        def test_connect_by_process(self):
+            """Test that connect_() works with a process"""
+            app1 = Application()
+            app1.start(bundle_id='com.yourcompany.send-keys-test-app')
+            app_conn = Application()
+            app_conn.connect(process=app1.process_id)
+            self.assertEqual(app1.process_id, app_conn.process_id)
+            app1.kill()
+
+        def test_connect_by_bundle(self):
+            """Test that connect_() works with a bundle"""
+            app1 = Application()
+            app1.start(bundle_id='com.yourcompany.send-keys-test-app')
+            app_conn = Application()
+            app_conn.connect(bundle='com.yourcompany.send-keys-test-app')
+            self.assertEqual(app1.process_id, app_conn.process_id)
+            app1.kill()
+
+        def test_connect_by_name(self):
+            app1 = Application()
+            app1.start(bundle_id='com.yourcompany.send-keys-test-app')
+            app_conn = Application()
+            app_conn.connect(name='send_keys_test_app')
+            self.assertEqual(app1.process_id, app_conn.process_id)
+            app1.kill()
```

Comment:
All `app = Application()` should be placed in `setUp` and all `app.kill()` should be in `tearDown()` tests that don't fit should be moved into a separate test suite